### PR TITLE
Migrate ssh exception -> config error exception

### PR DESCRIPTION
**Pull request by akashkulk**
Couple of exception handling changes:
- Migrate ssh exception -> config error exception
- Unnest in `IntegrationRunner` to determine whether an exception is a config error. In the codebase, many exceptions are caught and re-packaged as `RuntimeException`, so we have to be grab the root exception to check if it is a `config-error`

### CODE REVIEW

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java [hunk 1]
```diff
@@ -179,18 +183,30 @@ private void runInternal(final IntegrationConfig parsed) throws Exception {
...
   }
 
-  private boolean isConfigError(final Exception e) {
+  private Throwable getRootThrowable(final Exception e) {
+    Throwable current = e;
+    while (current != null) {
+      if (isConfigError(current)) {
+        return current;
+      } else {
+        current = current.getCause();
```
Comment (rodireich) [inline airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:192]: Like peeling an onion..

**Review by rodireich — APPROVED** [review]

File: airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java (hunk 1 above)
Comment (ryankfu) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:192]: May be able to simple use google common's library to do something like this

https://github.com/spotify/helios/blob/0b2855955dc6f3268e772252d84861082a516b6e/helios-services/src/main/java/com/spotify/helios/agent/HealthCheckerFactory.java#L153-L159

Comment (akashkulk) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:192]: Good find! Unfortunately, `Throwables.getRootCause` gets the root exception. Here, we actually want the root exception that is a possible configexception. The nuance here is that we *could* catch some `SQLException`, repackage it as a `ConfigErrorException` which then is repackaged up the stack as a `RuntimeException`.

We actually want the `ConfigErrorException`, so I think we have to manually unnest this

Comment (ryankfu) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:192]: I see, then can we add a javadoc comment that indicates this looks for the first instance of a `ConfigErrorException`? Also curious what happens if the Exception does not contain a `ConfigErrorException`, it seems would it be returning `null`, right?

Comment (akashkulk) [reply airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/base/IntegrationRunner.java:192]: Ahhh good catch :(
Will just return the original exception if there is no config error. Also updated the comments

**Review by ryankfu — APPROVED** [review]
LGTM, thanks for making sure we're not the ones last min causing a NPE

**Pull request merged**